### ISSUE TITLE
Class stats is not accurate

### ISSUE THREAD

**Issue by wz2k** [wz2k/pe#10]
When I only have 2 graded students with 10 and 30 marks.

![image.png](https://raw.githubusercontent.com/wz2k/pe/main/files/d9aa67a0-0b76-49b2-89b1-e45e8db64fba.png)

The mean is 30 which is wrong

![image.png](https://raw.githubusercontent.com/wz2k/pe/main/files/a32fc802-c946-4c08-bd9e-81fd3396d928.png)

**Comment by nus-pe-script** [comment]
# Team's Response
Hello, thanks for the report! However, would it be possible to list the steps done to obtain this output? We are unable to reproduce the bug by following the steps below:
1. Open TAA (fresh instance without any saved data)
1. `add_asgn n/Lab 1 hi`
2. `grade n/Lab 1 hi i/2 m/10`
3. `grade n/Lab 1 hi i/3 m/30`

This is the graph and statistics obtained, with the correct mean of 20:
![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2223S2/pe/master/files/1feae459-2a85-4d1c-b04b-dac1061f7441.png)

Our guess is that it could be possible that there were additional graded submissions that were unintentionally left out during the testing.![](url)
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.CannotReproduce`]

- [x] I disagree

**Reason for disagreement:** There are 2 different bugs that I posted (one is this and the other is about the grade command not working) and your team cannot reproduce both of them.
Took me awhile to find out, but turns out there is an indirect cause. This is actually caused by the `edit_student` command and both the bugs are linked to this command so I would be accepting the other issue as cannot reproduced and writing the combined response in this issue (as directed by prof).

Steps to reproduce.

1. Start with the default data
2. enter `add_asgn n/a`
3. enter `edit_student 1 cl/`
4. enter `grade n/a i/2 m/30 late/`

You should get the bug in step 4. In the interface nothing will happen when you press enter on the step 4 command, but checking the console will give you the following error.
![image.png](https://raw.githubusercontent.com/wz2k/pe/main/files/070ff52b-e186-47ec-9aa6-73e93514ca1f.png)

5. enter `grade n/a i/3 m/70 late/` (Nothing visible should happen again on the UI)
6. enter `class_stats st/grades as/a`

You should get the following. As you can see, even when on the UI all students are ungraded but there is some grade data present.

![image.png](https://raw.githubusercontent.com/wz2k/pe/main/files/5d4fc0e0-b239-42ed-a5ac-bd51bd2d2425.png)

After editing classes of students, the user will no longer be able to use the grade command properly. However, the bug does not stop there. Even though the grade command did not appear to do anything from the user's perspective, the grade is actually saved somewhere and this will cause the bug stated in this issue where class statistics are affected by the 'hidden' grades. The `ungrade` command would also appear to not do anything when entered.

What this means is that whenever a user edits classes of a student, it is possible to completely break grading and trying to grade would introduce hidden grades that the user is unable to see or confirm that the data is correct. The grades will silently be in the app and cause inaccuracies in statistics. A user might not know how to fix this which means that the app is considered permanently broken for them and would see no value in using it anymore. It is not easy for the user to identify the cause of the bug, just like how I originally reported the grade and class_stats command to not be working so even after fixing it, maybe by downloading a fresh instance, they will be hit by it again. To add on, grading is a serious matter for TAs and the grade data must be consistent and accurate.

To sum up the editing classes breaks certain commands in the app and cause inconsistent data, it is also difficult for the user to identify and avoid this bug, hence I would like to upgrade the severity of this issue to **HIGH**.

<catcher-end-of-segment><hr>
